h2o-core/src/main/java/hex/grid/HyperSpaceWalker.java

```diff
@@ -361,13 +432,18 @@ public MP nextModelParameters(Model previousModel) {
           _currentHyperparamIndices = _currentHyperparamIndices != null ? nextModelIndices(_currentHyperparamIndices) : new int[_hyperParamNames.length];
           if (_currentHyperparamIndices != null) {
             // Fill array of hyper-values
-            Object[] hypers = hypers(_currentHyperparamIndices, new Object[_hyperParamNames.length]);
-            // Get clone of parameters
-            MP commonModelParams = (MP) _params.clone();
-            // Fill model parameters
-            MP params = getModelParams(commonModelParams, hypers);
+            Object[] permutation = permutation(_currentHyperparamIndices, new Object[_hyperParamNames.length]);
+
+            if (_filterFunctions == null || !permutationIsSkipped(permutation, _hyperParamNames, _filterFunctions)) {
```

Comment:
I would recommend to handle the `_filterFunctions == null` case in `permutationIsSkipped`:  also used in RandomWalker

Reply:
@seb-h2o the intention was to more explicitly show  execution path for backward compatibility scenario. But if you insist I will move it inside. Good point anyway, thanks